### ISSUE TITLE
jest-expo-enzyme TypeError: _NativeUIManager.default.getConstants is not a function

### ISSUE THREAD

**Issue by jaimebiernaski** [expo/expo#7135]
### Environment

  Expo CLI 3.13.1 environment info:
    System:
      OS: macOS 10.15.3
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 12.3.1 - /usr/local/bin/node
      Yarn: 1.16.0 - /usr/local/bin/yarn
      npm: 6.10.3 - /usr/local/bin/npm
    IDEs:
      Android Studio: 3.4 AI-183.6156.11.34.5522156
      Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
    npmPackages:
      @types/react: ~16.9.0 => 16.9.21
      @types/react-native: ~0.60.23 => 0.60.31
      expo: ~36.0.0 => 36.0.2
      react: ~16.9.0 => 16.9.0
      react-native: https://github.com/expo/react-native/archive/sdk-36.0.0.tar.gz => 0.61.4
    npmGlobalPackages:
      expo-cli: 3.13.1

Target: iOS, Android

### Steps to Reproduce

1. `expo init` with TypeScript template
2. `yarn add jest-expo-enzyme --dev`
3. Add the following config to package.json:

```
"scripts": {
  "test": "jest",
},
"jest": {
  "preset": "jest-expo-enzyme"
}
```
3. Create a file App.test.tsx with the content

```
it('works', () => {
  expect(1).toBe(1);
});
```
4. Exec `yarn test`

### Expected Behavior

Run without error

### Actual Behavior
**Warning**
react-native/jest-preset contained different transformIgnorePatterns than expected

**FAIL**
The test passes for Web but fails for iOS & Android
    `TypeError: _NativeUIManager.default.getConstants is not a function`
```
➜  test git:(master) ✗ yarn test
yarn run v1.16.0
$ jest
react-native/jest-preset contained different transformIgnorePatterns than expected
 FAIL   iOS  ./App.test.tsx
  ● Test suite failed to run

    TypeError: _NativeUIManager.default.getConstants is not a function

      at getConstants (node_modules/react-native/Libraries/ReactNative/PaperUIManager.js:28:48)
      at Object.<anonymous> (node_modules/react-native/Libraries/ReactNative/PaperUIManager.js:129:15)
      at Object.<anonymous> (node_modules/react-native/Libraries/ReactNative/UIManager.js:37:7)
      at Object.user:/private/tmp/expotTest/test/node_modules/react-native/Libraries/ReactNative/UIManager.js: (node_modules/jest-expo/src/preset/setup.js:205:25)
      at Object.<anonymous> (node_modules/react-native/Libraries/Utilities/deprecatedPropType.js:13:19)
      at Object.<anonymous> (node_modules/react-native/Libraries/DeprecatedPropTypes/DeprecatedTransformPropTypes.js:15:28)
      at Object.<anonymous> (node_modules/react-native/Libraries/DeprecatedPropTypes/DeprecatedImageStylePropTypes.js:15:38)
      at Object.<anonymous> (node_modules/react-native/Libraries/Components/View/ReactNativeStyleAttributes.js:13:39)
      at Object.<anonymous> (node_modules/react-native/Libraries/StyleSheet/StyleSheet.js:13:36)
      at Object.get StyleSheet [as StyleSheet] (node_modules/react-native/Libraries/react-native/react-native-implementation.js:273:12)
      at Object.<anonymous> (node_modules/jest-expo-enzyme/src/serializer.ts:56:23)
      at Object.<anonymous> (node_modules/jest-expo-enzyme/src/setupEnzyme.native.ts:8:1)
          at Array.forEach (<anonymous>)

 FAIL   Android  ./App.test.tsx
  ● Test suite failed to run

    TypeError: _NativeUIManager.default.getConstants is not a function

      at getConstants (node_modules/react-native/Libraries/ReactNative/PaperUIManager.js:28:48)
      at Object.<anonymous> (node_modules/react-native/Libraries/ReactNative/PaperUIManager.js:132:12)
      at Object.<anonymous> (node_modules/react-native/Libraries/ReactNative/UIManager.js:37:7)
      at Object.user:/private/tmp/expotTest/test/node_modules/react-native/Libraries/ReactNative/UIManager.js: (node_modules/jest-expo/src/preset/setup.js:205:25)
      at Object.<anonymous> (node_modules/react-native/Libraries/Utilities/deprecatedPropType.js:13:19)
      at Object.<anonymous> (node_modules/react-native/Libraries/DeprecatedPropTypes/DeprecatedTransformPropTypes.js:15:28)
      at Object.<anonymous> (node_modules/react-native/Libraries/DeprecatedPropTypes/DeprecatedImageStylePropTypes.js:15:38)
      at Object.<anonymous> (node_modules/react-native/Libraries/Components/View/ReactNativeStyleAttributes.js:13:39)
      at Object.<anonymous> (node_modules/react-native/Libraries/StyleSheet/StyleSheet.js:13:36)
      at Object.get StyleSheet [as StyleSheet] (node_modules/react-native/Libraries/react-native/react-native-implementation.js:273:12)
      at Object.<anonymous> (node_modules/jest-expo-enzyme/src/serializer.ts:56:23)
      at Object.<anonymous> (node_modules/jest-expo-enzyme/src/setupEnzyme.native.ts:8:1)
          at Array.forEach (<anonymous>)

 PASS   Web  ./App.test.tsx (12.846s)

Test Suites: 2 failed, 1 passed, 3 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        14.361s
Ran all test suites in 3 projects.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by jaimebiernaski** [comment]
Hi,

It was missing the TypeScript configuration for Jest.

- Install the types for jest `@types/jest`

- Install the TypeScript preprocessor for jest `ts-jest` which allows jest to transpile TypeScript on the fly and have source-map support built in.

- Add the following `jest.config.js` file to the root of your project:

```
module.exports = {
  roots: ["<rootDir>/src"],
  transform: {
    "^.+\\.tsx?$": "ts-jest"
  },
  testRegex: "(/__tests__/.*|(\\.|/)(test|spec))\\.(jsx?|tsx?)$",
  moduleFileExtensions: ["ts", "tsx", "js", "jsx", "json", "node"]
};
```

**Comment by jaimebiernaski** [comment]
I've tried to set up according to the reference of the [TypeScript Book](https://github.com/basarat/typescript-book/blob/master/docs/testing/jest.md). 

- Add the following` jest.config.js` file to the root of your project:

```
module.exports = {
  "roots": [
    "<rootDir>/src"
  ],
  "testMatch": [
    "**/__tests__/**/*.+(ts|tsx|js)",
    "**/?(*.)+(spec|test).+(ts|tsx|js)"
  ],
  "transform": {
    "^.+\\.(ts|tsx)$": "ts-jest"
  },
}
```
The test above passes, but when I try to test the App Component I get the error:

```
ts-jest[versions] (WARN) Version 24.9.0 of jest installed has not been tested with ts-jest. If you're experiencing issues, consider using a supported version (>=25.0.0 <26.0.0). Please do not report issues in ts-jest if you are using unsupported versions.
 FAIL  src/App.test.tsx
  ● Test suite failed to run

    Jest encountered an unexpected token

    This usually means that you are trying to import a file which Jest cannot parse, e.g. it's not plain JavaScript.

    By default, if Jest sees a Babel config, it will use that to transform your files, ignoring "node_modules".

    Here's what you can do:
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/en/configuration.html

    Details:

    /Users/jaime/Workspace/audem/app/src/App.test.tsx:7
        var wrapper = enzyme_1.shallow(<App_1.default />);
                                       ^

    SyntaxError: Unexpected token <

      at ScriptTransformer._transformAndBuildScript (node_modules/@jest/transform/build/ScriptTransformer.js:537:17)
      at ScriptTransformer.transform (node_modules/@jest/transform/build/ScriptTransformer.js:579:25)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        3.276s
Ran all test suites.
error Command failed with exit code 1.
```

**Comment by expjess** [comment]
Hey @jaimebiernaski - I think your issue got might've gotten cut off right after the Repro heading; can you maybe add it again in the comments if you have a minimal repro example? Thanks!

**Comment by jaimebiernaski** [comment]
Hi, follows the Repro

1. `git clone https://github.com/jaimebiernaski/repro_issue_7135.git`

2. `cd repro_issue_7135`

3. `yarn install`

4. `yarn test`

**Comment by andreicode** [comment]
Hello, 

I have the same issue. And I think I figured where it happens, It has to do with changes to UIManager in jest-expo setup in this file: https://github.com/expo/expo/blob/4d08ec514ab4307166a7b8002ac194a0a23cbf00/packages/jest-expo/src/preset/setup.js#L200-L208
This commits fixes the issue : 440ce7412db5d49ac2cab23a96532e760dcd3fb6 and 8d841e525e48974627b54f6e2c87297f9a180e73 by making **jest-expo-enzyme** use **jest-expo** _36.0.1_

Unfortunately I can't get **jest-expo-enzyme** to have **jest-expo** at _36.0.1_ with **expo** _36.0.2_. For some reason npm install will pull **jest-expo** @ _35.0.0_ 

I mean, I can force jest-expo-enzyme to use "jest-expo": "~36.0.1" by modifying node modules and then the test passes...

**Comment by cruzach** [comment]
I just published `jest-expo-enzyme` v1.0.0, could you give that a try and let me know if the problem is fixed?

**Comment by jaimebiernaski** [comment]
Hi @cruzach, 

I've just tested the version `jest-expo-enzyme` **v1.0.0** and the problem was fixed!

Thank you so much!
